### PR TITLE
uxmal: A ZeroBuf vocabulary and application for LOD and cull debugging

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
uxmal

File: examples/triply/vertexBufferNode.cpp [hunk 1]
```diff
@@ -82,14 +82,24 @@ void VertexBufferNode::setupTree(VertexData& data, const Index start,
...
-    // take the bounding spheres returned by the children
-    const BoundingSphere& sphere1 = _left->updateBoundingSphere();
-    const BoundingSphere& sphere2 = _right->updateBoundingSphere();
+    _left->updateBounds();
+    _right->updateBounds();
+
+    // compute enclosing box
+    const auto& box1 = _left->getBoundingBox();
+    const auto& box2 = _right->getBoundingBox();
+    _boundingBox[0][0] = std::min(box1[0][0], box2[0][0]);
```
Comment (tribal-tec) [inline examples/triply/vertexBufferNode.cpp]: merge()?

File: examples/uxmal/uxmal.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    return eq::Vector3f(eq::Matrix4f(m, m + 16) * eq::Vector4f(x, y, z, 1.f));
+}
+static const std::vector<eq::Vector3f> _colors{{1, 1, 1}, {1, 0, 0}, {0, 1, 0},
+                                               {0, 0, 1}, {1, 1, 0}, {1, 0, 1},
+                                               {0, 1, 1}};
+}
+
+struct Box
+{
+    explicit Box(const eq::uint128_t& i, const Frustum& f)
```
Comment (tribal-tec) [inline examples/uxmal/uxmal.cpp]: not explicit

File: examples/uxmal/uxmal.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                  _xfm(f.getRight() * f.getFar() / f.getNear(),
+                       f.getTop() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView()),
+                  _xfm(f.getLeft() * f.getFar() / f.getNear(),
+                       f.getTop() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView())}
+    {
+    }
+
+    explicit Box(const eq::uint128_t& i, const AABB& b)
```
Comment (tribal-tec) [inline examples/uxmal/uxmal.cpp]: dito

File: examples/uxmal/uxmal.cpp [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                              [this](const void* data, const size_t size) {
+                                  _handle<AABB>(data, size);
+                              });
+    }
+    virtual ~Renderer() {}
+protected:
+    bool init(co::Object*) final
+    {
+        glEnable(GL_LINE_SMOOTH);
+        glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
```
Comment (tribal-tec) [inline examples/uxmal/uxmal.cpp]: funny, I looked that up too recently. Turns out it does not work (well) on some machines like the cluster iirc. anti-aliased lines are still impossible in 2017 :p

File: examples/triply/vertexBufferNode.cpp (hunk 1 above)
Comment (eile) [inline examples/triply/vertexBufferNode.cpp]: Not an AABB, but a Vector3f[2]. Might clean it up with another improvement in progress.

File: examples/uxmal/uxmal.cpp (hunk 3 above)
Comment (eile) [inline examples/uxmal/uxmal.cpp]: Thanks, this did hide when adding the second param. ;)

File: examples/uxmal/uxmal.cpp (hunk 4 above)
Comment (eile) [inline examples/uxmal/uxmal.cpp]: (fp). Works well on my local machine, see screenshot. Will tweak this later.

File: examples/triply/vertexBufferNode.cpp (hunk 1 above)
Comment (eile) [inline examples/triply/vertexBufferNode.cpp]: Done in 0528cbb

**Comment by eile** [comment]
Before/after cull results for 0528cbb:
sphere
aabb

**Pull request merged**